miner/scroll_worker.go

```diff
@@ -372,7 +372,7 @@ func (w *worker) mainLoop() {
 		select {
 		case <-w.startCh:
 			idleTimer.UpdateSince(idleStart)
-			if w.isRunning() {
+			if w.isRunning() && w.chainConfig.Scroll.UseZktrie {
```

Comment:
For `l2geth-zktrie`, will `isRunning` become false after the Euclid timestamp?

Reply:
No, it will keep behaving like it used to. (ie `true` if `--mine` is set)

Reply:
Won't `checkHeadRowConsumption` fail on `l2geth-zktrie` after Euclid? (It seems like we don't disable it.)

Reply:
CCC just skips all post-Euclid blocks, so it is fine. https://github.com/scroll-tech/go-ethereum/pull/1104/files#diff-a7730ed484f9261277ae9380669a09437e556095c9992d4be6266d6b9c745142R101-R105